src/main/java/com/hms_networks/americas/sc/extensions/time/LocalTimeOffsetCalculator.java

```diff
@@ -56,65 +30,72 @@ public static synchronized long getLocalTimeOffsetMilliseconds() {
 
     // Parse local time offset value file and store/return
-    timeOffsetMilliseconds = parseOffsetFromLocalTime();
+    timeOffsetMilliseconds = parseLocalTimeFile();
     return timeOffsetMilliseconds;
   }
 
   /**
-   * Generates a local time offset HTML files and creates an HTTP connection to that page to
-   * generate a local time offset result file. The HTTP connection is authenticated using the
-   * default Ewon device credentials, or updated device credentials that can be set via {@link
-   * #setDeviceUsername(String)} and {@link #setDevicePassword(String)}.
+   * Generates an exported local time result file using the script expression export block
+   * descriptor call specified by {@link TimeLibConstants#TIME_OFFSET_LOCAL_TIME_EBD_PREFIX} and the
+   * file path specified by {@link TimeLibConstants#TIME_OFFSET_DIRECTORY_PATH} and {@link
+   * TimeLibConstants#TIME_OFFSET_LOCAL_TIME_FILE_NAME}.
    *
-   * @throws IOException if unable to write local time offset HTML file
-   * @throws EWException if unable to create HTTP connection to local time offset HTML file (check
-   *     device username and password)
+   * @throws IOException if unable to perform script expression export block descriptor call or
+   *     unable to write script expression export block descriptor call result file
    */
-  private static synchronized void generateLocalTimeOffsetFiles() throws IOException, EWException {
-    // Write local time offset HTML file
-    FileAccessManager.writeStringToFile(
-        TimeLibConstants.TIME_OFFSET_HTML_FILE_NAME,
-        TimeLibConstants.TIME_OFFSET_HTML_FILE_CONTENTS);
+  private static synchronized void exportLocalTimeResultFile() throws IOException {
+    // Create folder if it does not exist so export block descriptor call can write to it
+    File localTimeResultFolder = new File(TimeLibConstants.TIME_OFFSET_DIRECTORY_PATH);
+    if (!localTimeResultFolder.exists()) {
+      localTimeResultFolder.mkdirs();
+    }
 
-    // Load and write local time offset value file
-    String httpUserCredentialsAndServer =
-        deviceUsername + ":" + devicePassword + "@" + TimeLibConstants.LOOPBACK_IP_ADDRESS;
-    ScheduledActionManager.GetHttp(
-        httpUserCredentialsAndServer,
-        TimeLibConstants.TIME_OFFSET_RESULT_FILE_NAME,
-        TimeLibConstants.TIME_OFFSET_HTML_FILE_NAME);
+    // Build full export block descriptor call and file path
+    final String exportBlockDescriptorCall =
+        TimeLibConstants.TIME_OFFSET_LOCAL_TIME_EBD_PREFIX
+            + TimeLibConstants.TIME_OFFSET_LOCAL_TIME_FILE_NAME;
+    final String exportBlockDescriptorFilePath =
+        FileConstants.FILE_URL_PREFIX
+            + TimeLibConstants.TIME_OFFSET_DIRECTORY_PATH
+            + TimeLibConstants.TIME_OFFSET_LOCAL_TIME_FILE_NAME;
+
+    // Run export block descriptor call to get local time
+    Exporter exporter = new Exporter(exportBlockDescriptorCall);
+    exporter.ExportTo(exportBlockDescriptorFilePath);
```

Comment:
Have you tried to read from EBD stream?   You may not need to save results to file. The JAVA API docs show a read and available methods for the Export class instances.

Reply:
OR will an instance of SysControlBlock be a more elegant solution.

Reply:
This isn't something that is available via SysControlBlock as far as I'm aware

Reply:
You are right about SysControlBlock. 

just tested 
``` java
        byte[] results = new byte[50];
        String EBD = "$dtSE$se\"Time$\"";
        System.out.println("testing " + EBD);
        System.out.println(results);
        try{

        Exporter exporter = new Exporter(EBD);
        int available= exporter.available();
        System.out.println("available " + available);
        int res = exporter.read(results, 0 , 50);
        String resultStr = new String(results);
        System.out.println(resultStr);

        System.out.println("return value " + res);
        }catch(IOException e) {
            System.out.println(e.getMessage());
        }
```

![image](https://user-images.githubusercontent.com/72526279/149165914-3b09c6e1-b4b2-42f8-bca0-fd6939ffcddc.png)

Reply:
I've implemented a string utility to read a given input stream and return a string. I took a look at the exporter class itself extends input stream, so all that was required was to pass the exporter to that. It's been implemented